### PR TITLE
Rework documentation

### LINKED ISSUE #47: Change CONFIG_REFERENCE.md to be a manpage
scdoc is probably the way to go.

### PR DESCRIPTION
**Pull request by foxcpp**
CONFIG_REFERENCE.md is now maddy.conf(5) man page.
README.md is cleaned up.
Also, add maddy(1) man page.

Close #47.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,151 +2,48 @@
 
 [![builds.sr.ht status](https://builds.sr.ht/~emersion/maddy.svg)](https://builds.sr.ht/~emersion/maddy?)
 
-Fast, cross-platform mail server.
+Simple, fase IMAP+SMTP mail server.
```
Comment (emersion) [inline README.md]: Typo

File: README.md [hunk 2]
```diff
@@ -2,151 +2,48 @@
...
+IMAP server functionality in one application.
+
+**⚠️ Warning:** maddy is in development, many important features are missing, there
+are bugs and performance can be bad.
+
+## Getting started
+
+### Installation
+
+You need Go 1.11 or newer. C compiler is required for SQLite3 storage support
```
Comment (emersion) [inline README.md]: "A C compiler is …"

File: maddy.1.scd [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	statedir in configuration.
+
+*MADDYLIBEXEC*
+	Can be used to override libexec directory. Does have higher prirotiy than
+	libexecdir in configuration.
+
+# AUTHORS
+
+Maintained by Simon Ser <contact@emersion.fr> and Max Mazurov
+<fox.cpp@disroot.org>, who is assisted by other open-source contributors. For
```
Comment (emersion) [inline maddy.1.scd]: s/is/are/

File: maddy.1.scd [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+more information about maddy development, see
+https://github.com/emersion/maddy.
+
+# SEE ALSO
+
+*maddy.conf*(5), *maddy-pam-helper*(1), *maddy-shadow-helper*(1)
```
Comment (emersion) [inline maddy.1.scd]: These won't be in `PATH` and will be internal so they probably won't get a manpage

**Comment by emersion** [comment]
Needs rebase

**Comment by foxcpp** [comment]
Rebased.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**